### PR TITLE
Issue 75 Parallelization of CFL

### PR DESCRIPTION
**Pull request by ChristianVonMach**
Closing in #75 mentioned problems

### CODE REVIEW

**Comment by ChristianVonMach** [comment]
```
Analysis::calc_CFL(Field *, Field *, Field *, double):
    341, Generating Tesla code
        341, Generating reduction(max:cfl_max)
        346, #pragma acc loop gang, vector(128) /* blockIdx.x threadIdx.x */
    341, Generating implicit copyin(d_v,d_u) [if not already present]
         Generating implicit copy(cfl_max) [if not already present]
         Generating implicit copyin(innerList[:],d_w) [if not already present]
    346, Accelerator restriction: size of the GPU copy of d_u,d_w,d_v is unknown
```
compiler output for CFL calculation with

`docker run -i --rm -v (pwd):/host_pwd -w /host_pwd -u (id -u user):(id -g user) artss_docker sh -c "./compile.sh --jobs 4 --gb"`
```
-- Using PGI 19.10.0
-- Release flags:  -std=c++17 -Mnollvm -std=c++17 ,  -fast -O3 -DNDEBUG -O3 -fast -w -DNDEBUG
-- 
-- Platform: 
--     Host: Linux4.9.237x86_64
--     CMake: 3.10.2
--     CMake generator: Unix Makefiles
--     CMake build tool: /usr/bin/make
--     Configuration: Release
```

File: src/TimeIntegration.cpp [hunk 1]
```diff
@@ -170,7 +170,11 @@ void TimeIntegration::run() {
 #pragma acc update host(d_S_T[:bsize]) wait    // all in one update does not work!
 
             visual->visualise(m_solver, t_cur);
-            if (adaption->is_data_extraction_before_enabled()) adaption->extractData(adaption->get_before_name(), adaption->get_before_height(), t_cur);
+            if (adaption->is_data_extraction_before_enabled())
```
Comment (LinhWuerzburger) [inline src/TimeIntegration.cpp:173]: if you change the one line to two lines, you should add { } at least

File: src/TimeIntegration.cpp [hunk 2]
```diff
@@ -196,18 +196,20 @@ void TimeIntegration::run() {
             // update
             adaption->run(t_cur);
 #ifndef BENCHMARKING
-            if (adaption->is_data_extraction_after_enabled()) adaption->extractData(adaption->get_after_name(), adaption->get_after_height(), t_cur);
+            if (adaption->is_data_extraction_after_enabled())
```
Comment (LinhWuerzburger) [inline src/TimeIntegration.cpp:199]: same here

**Review by LinhWuerzburger — CHANGES_REQUESTED** [review]
the code structure changes have to be addressed first.
I didn't control the parallelisation, but it looks fine

File: src/TimeIntegration.cpp (hunk 1 above)
Comment (ChristianVonMach) [reply src/TimeIntegration.cpp]: done in ee47e67

File: src/TimeIntegration.cpp (hunk 2 above)
Comment (ChristianVonMach) [reply src/TimeIntegration.cpp]: done in ee47e67

**Review by LinhWuerzburger — APPROVED** [review]

**Pull request merged**